File: Java/src/main/java/QuestionBank/Solution001_100/Solution53.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution53 {
    public int maxSubArray(int[] nums) {
        int sum = nums[0];
        int max = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (sum < 0){
                sum = nums[i];
            }else{
                sum += nums[i];
                max = Math.max(max, sum);
            }
        }
        return max;
    }


    public static void main(String[] args) {
        Solution53 solution53 = new Solution53();
    }
}

File: C++/src/Solution1_100/Solution53.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution53.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include "vector"
using namespace std;


int maxSubArray(vector<int>& nums) {
  int maxVal = nums[0];
  int sum = nums[0];
  for (int i = 1; i < nums.size(); ++i) {
    sum += nums[i];
    if (sum < 0){
      sum = 0;
    }
    maxVal = max(maxVal, sum);
  }
  return maxVal;
}

